src/librustc_mir/util/pretty.rs

```diff
@@ -127,10 +130,12 @@ fn dump_matched_mir_node<'tcx, F>(
 {
     let _: io::Result<()> = try {
         let mut file = create_dump_file(tcx, "mir", pass_num, pass_name, disambiguator, source)?;
-        writeln!(file, "// MIR for `{}`", node_path)?;
-        writeln!(file, "// source = {:?}", source)?;
-        writeln!(file, "// pass_name = {}", pass_name)?;
-        writeln!(file, "// disambiguator = {}", disambiguator)?;
+        write!(file, "// MIR for `{}", node_path)?;
```

Comment:
`node_path` should be renamed to `def_path` just to reflect that it came from `def_path_str`.
Or even better, that bit of code could be inlined here.

Any chance we can use the `Instance` printing (inside `with_forced_impl_filename_line` ofc)? Or is it pointless as we don't optimize shims? (Do we? Do we not? I can't tell)

Reply:
I don't think we dump shims, at least I've never encountered any, I'll look into it. Just doing the inlining right here for this PR.

Reply:
`-Zdump-mir` dumps shims. I made it do that some time... last year or so.

For `--emit mir`, there's https://github.com/rust-lang/rust/issues/58102.